src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -574,10 +574,16 @@ private function registerAssetsConfiguration(array $config, ContainerBuilder $co
     {
         $loader->load('assets.xml');
 
+        $removeGitStrategy = true;
```

Comment:
So, let's reverse this logic and define dynamically the `assets.git_version_strategy` service when it's the value of the `version_strategy` config option. And also, let's compute the git hash inline in the FrameworkExtension. (which means removing GitVersionStrategy and the patch on the xml file).
Btw, I think the git process should have its working directory forced to `kernel.root_dir`.